crates/nu-command/src/system/inspect.rs

```diff
@@ -0,0 +1,369 @@
+use nu_engine::{eval_expression, CallExt};
+use nu_protocol::ast::{Argument, Block, Call, Expr, Expression};
+use nu_protocol::engine::{Closure, Command, EngineState, Stack};
+use nu_protocol::{
+    Category, Example, IntoInterruptiblePipelineData, PipelineData, ShellError, Signature, Span,
+    SyntaxShape, Type, Value,
+};
+
+#[derive(Clone)]
+pub struct Inspect;
+
+impl Command for Inspect {
+    fn name(&self) -> &str {
+        "inspect"
+    }
+
+    fn usage(&self) -> &str {
+        "Inspect closure contents."
+    }
+
+    fn signature(&self) -> nu_protocol::Signature {
+        Signature::build("inspect")
+            .required(
+                "closure",
+                SyntaxShape::Closure(Some(vec![SyntaxShape::Any])),
+                "the closure to run",
+            )
+            .input_output_types(vec![(Type::Any, Type::Any), (Type::Nothing, Type::Any)])
+            .allow_variants_without_examples(true)
+            .category(Category::System)
```

Comment:
`Category::Debug` as soon as that is available through the other PRs

Reply:
Agreed